tlib/src/net.cpp

```diff
@@ -158,8 +159,8 @@ std::expected<tlib::packet> tlib::wait_for_packet(size_t socket_fd) {
                  : [socket] "g"(socket_fd), [buffer] "g"(reinterpret_cast<size_t>(buffer))
                  : "rax", "rbx", "rcx");
 
+    free(buffer);
```

Comment:
This is wrong, because `buffer` is returned by the system call as`payload` and therefore is used then and returned by the function.